### PR TITLE
chore(IDX): updates to bot check script

### PR DESCRIPTION
**Pull request by cgundy**
There are still some issues with the script which I haven't fully debugged. These changes explicitly fail the script and add more debugging messages.

### CODE REVIEW

File: reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py [hunk 1]
```diff
@@ -22,8 +22,13 @@ def get_changed_files(merge_base_sha: str, branch_head_sha: str) -> list[str]:
...
     commit_range = f"{merge_base_sha}..{branch_head_sha}"
     result = subprocess.run(
-        ["git", "diff", "--name-only", commit_range], stdout=subprocess.PIPE, text=True
+        ["git", "diff", "--name-only", commit_range],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        text=True,
     )
+    if result.returncode != 0:
+        raise RuntimeError(f"git diff failed: {result.stderr}")
```
Comment (basvandijk) [inline reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py:31]: Wouldn't `result.stderr` be `None` in this case?

I believe you need to set `capture_output=True` and no set `stderr=subprocess.DEVNULL`.

File: .github/workflows/repo_policies.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
       - name: Bot Checks
         id:  bot-checks
         run: |
+          set -eufo pipefail
```
[the remark this answers is not in the document]
Comment (cgundy) [reply .github/workflows/repo_policies.yml:24]: Ah good catch, I just copied this from an existing config.

File: reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py (hunk 1 above)
Comment (cgundy) [reply reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py]: Thanks, admittedly I still don't know what best practice is for calling `subprocess.run`. Is this all that's needed?

Comment (basvandijk) [reply reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py]: I don't know if there's a best practise. It just depends on what behaviour you want. I think just read [the docs](https://docs.python.org/3/library/subprocess.html#subprocess.run) carefully and convince yourself that it's correct. Then do test it of course.

**Review by basvandijk — APPROVED** [review]

**Pull request merged**